### PR TITLE
Revamp comment interface

### PR DESCRIPTION
**Pull request by laptou**
- Make the comment area look pretty.

Screenshot 2020-06-03 at 14 18 26 - Display 1

### CODE REVIEW

File: portfolio/src/main/client/src/components/controls/labeled-input.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import {htmlElement, htmlFragment} from '@src/util/html';
+
+export interface LabeledInputOptions {
```
Comment (jessicaslaughter) [inline portfolio/src/main/client/src/components/controls/labeled-input.ts:3]: add a JSDoc comment describing this interface

File: portfolio/src/main/client/src/components/controls/labeled-input.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+   */
+  soft?: boolean;
+  name?: string;
+  className?: string;
+  type?: 'text' | 'number' | 'email' | 'textarea';
+
+  value?: string;
+}
+
+export const LabeledInput =
```
Comment (jessicaslaughter) [inline portfolio/src/main/client/src/components/controls/labeled-input.ts:48]: add a JSDoc comment describing what this is as well

**Review by jessicaslaughter — CHANGES_REQUESTED** [review]

File: portfolio/src/main/client/src/components/controls/labeled-input.ts (hunk 1 above)
Comment (laptou) [reply portfolio/src/main/client/src/components/controls/labeled-input.ts:3]: The interface itself seems self-explanatory, but I added documentation for all of its members.

**Review by jessicaslaughter — APPROVED** [review]

File: portfolio/src/main/client/src/components/controls/labeled-input.ts (hunk 1 above)
Comment (jpnickolas) [reply portfolio/src/main/client/src/components/controls/labeled-input.ts:3]: Interfaces still need jsdocs. Anything exported should have a jsdoc.

File: portfolio/src/main/client/src/components/controls/labeled-input.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+ * empty, the label may disappear and be used as a placeholder instead.
+ */
+export const LabeledInput =
+  ({
+    name, id, className, label, type, value, soft,
+  }: LabeledInputOptions): Node[] => {
+    const labelEl: HTMLLabelElement =
+      htmlElement`<label for="${id}" class="input-label">${label}</label>`;
+
+    if (soft !== false) labelEl.classList.add('soft');
```
Comment (jpnickolas) [inline portfolio/src/main/client/src/components/controls/labeled-input.ts:55]: nit: you should make optional fields disabled by default. It seems pretty unintuitive that providing {soft: true} and {} do the same things.

File: portfolio/src/main/client/src/components/controls/labeled-input.ts [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+    const inputEl: HTMLInputElement | HTMLTextAreaElement =
+      type === 'textarea' ?
+        htmlElement`<textarea id="${id}" 
+                              name="${name}" 
+                              placeholder="${label}">${value}</textarea>` :
+        htmlElement`<input id="${id}"
+                           class="${className}"
+                           name="${name}" 
+                           type="${type ?? 'text'}"
```
Comment (jpnickolas) [inline portfolio/src/main/client/src/components/controls/labeled-input.ts:66]: best practice is to use an "or" statement when providing default values for a nullable/falsey value:
`type || 'text'`

**Review by jpnickolas — CHANGES_REQUESTED** [review]

**Comment by laptou** [comment]
I accidentally merged this PR. I'm used to seeing the big green button that says "merge" and just immediately clicking it, forgetting that this PR is not based against `master`. There doesn't seem to be a way to reopen it, so I will just make a new one.

**Pull request merged**